### DISCUSSION TITLE
Automatic line-breaking and multi-column cells

### DISCUSSION THREAD

**Discussion by fefrei** [lvjr/tabularray#140] [Q&A]
`tabularray`'s `X` columns, fortunately, break lines automatically (and `tabularray` generally copes really well with multi-line cells).

However, I sometimes find myself having long multi-column cells (spanning `X` columns, typically). Unfortunately, these do not auto-break, causing the table to get wider than expected.

Here's a small example:

```latex
\documentclass{article}

\usepackage{lipsum}
\usepackage{tabularray}

\begin{document}
    \lipsum[1]

    \begin{tblr}{colspec={XXXX}, hlines, vlines}
        One & Two & Three & Four plus a lot of text that is nicely line-wrapped automatically \\
        One & \SetCell[c=2]{c} Two plus a lot of text that does not break automatically & & Four \\
    \end{tblr}

    \lipsum[2]
\end{document}
```

![image](https://user-images.githubusercontent.com/3078885/139229166-05320ecc-b7c8-4803-92de-22ca7935441a.png)

There obviously are many ways around this, but I only found ways that are somewhat hacky, requiring manually specifying widths or manually breaking lines. Is there some _nice_ way to handle these situations?

**Comment by lvjr** [comment]
You may use `hspan=minimal` option. See the manual.

**Reply by fefrei** [reply]
This _manual_ you mention holds many secrets. I shall consult it more! May this discussion guide more truth-seekers there.

(In all seriousness: Thanks, this is exactly what I was looking for, I just wasn't looking in the right places. Thanks for the pointer!)

**Reply by abstractionmage** [reply]
Thanks, but I would nonetheless expect tabularray to auto-break long multi-column cells by default when the table would otherwise exceed its specified `width`.
